tests/telemetry/telemetry_utils.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import pytest
+
+from pkg_resources import parse_version
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+TELEMETRY_PORT = 50051
+METHOD_GET = "get"
+METHOD_SUBSCRIBE = "subscribe"
+SUBSCRIBE_MODE_STREAM = 0
+SUBMODE_SAMPLE = 2
+
+
+def assert_equal(actual, expected, message):
+    """Helper method to compare an expected value vs the actual value.
+    """
+    pytest_assert(actual == expected, "{0}. Expected {1} vs actual {2}".format(message, expected, actual))
+
+
+def get_dict_stdout(gnmi_out, certs_out):
+    """ Extracts dictionary from redis output.
+    """
+    gnmi_list = []
+    gnmi_list = get_list_stdout(gnmi_out) + get_list_stdout(certs_out)
+    # Elements in list alternate between key and value. Separate them and combine into a dict.
+    key_list = gnmi_list[0::2]
+    value_list = gnmi_list[1::2]
+    params_dict = dict(zip(key_list, value_list))
+    return params_dict
+
+
+def get_list_stdout(cmd_out):
+    out_list = []
+    for x in cmd_out:
+        result = x.encode('UTF-8')
+        out_list.append(result)
+    return out_list
+
+
+def skip_201911_and_older(duthost):
+    """ Skip the current test if the DUT version is 201911 or older.
+    """
+    if parse_version(duthost.kernel_version) <= parse_version('4.9.0'):
+        pytest.skip("Test not supported for 201911 images. Skipping the test")
+
+
+def skip_arm_platform(duthost):
+    """ Skip the current test if DUT is arm platform.
+    """
+    platform = duthost.facts["platform"]
+    if 'arm' in platform:
```

Comment:
I'm not sure if we can use this key word to detect arm platform, can you confirm?
Another solution is to run "lscpu" or "uname -a" on DUT.

Reply:
Switched logic to only perform tests on platforms starting with x86_64